rskj-core/src/main/java/co/rsk/metrics/HashRateCalculatorImpl.java

```diff
@@ -18,48 +18,52 @@
 
 package co.rsk.metrics;
 
-import co.rsk.util.AccountUtils;
+import co.rsk.config.MiningConfig;
 import co.rsk.util.RskCustomCache;
 import org.ethereum.core.Block;
 import org.ethereum.db.BlockStore;
 import org.ethereum.db.ByteArrayWrapper;
 
 import java.math.BigInteger;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
 import java.util.Arrays;
-import java.util.concurrent.TimeUnit;
 import java.util.function.Predicate;
 
 /**
  * Created by mario on 05/08/2016.
  */
 public class HashRateCalculatorImpl implements HashRateCalculator {
 
-    private BlockStore blockStore;
+    private final BlockStore blockStore;
+    private final MiningConfig miningConfig;
+    private final RskCustomCache<ByteArrayWrapper, BlockHeaderElement> headerCache;
 
-    private RskCustomCache<ByteArrayWrapper, BlockHeaderElement> headerCache;
-
-    private final byte[] address;
-
-    public HashRateCalculatorImpl(BlockStore blockStore, AccountUtils accountUtils, RskCustomCache<ByteArrayWrapper, BlockHeaderElement> headerCache) {
+    public HashRateCalculatorImpl(BlockStore blockStore, MiningConfig miningConfig, RskCustomCache<ByteArrayWrapper, BlockHeaderElement> headerCache) {
```

Comment:
Instead of having this class receive an instance of MiningConfig maybe it's better to:
- Option 1. Have to classes to implement HashRateCalculator. One will be used when miningConfig.isMiningEnabled() and the other when not(miningConfig.isMiningEnabled()). Dependency Injection FTW :)
- Option 2. Similar to option 1 but using an abstract class with two subclasses.

Reply:
Done